src/iceberg/table_metadata.h

```diff
@@ -458,14 +467,40 @@ enum class ICEBERG_EXPORT MetadataFileCodecType {
   kGzip,
 };
 
-/// \brief Utility class for table metadata
-struct ICEBERG_EXPORT TableMetadataUtil {
+struct ICEBERG_EXPORT CodecTypeUtil {
+  /// \brief Returns the MetadataFileCodecType corresponding to the given string.
+  ///
+  /// \param name The string to parse.
+  /// \return The MetadataFileCodecType corresponding to the given string.
+  static Result<MetadataFileCodecType> CodecFromString(const std::string_view& name);
```

Comment:
```suggestion
  static Result<MetadataFileCodecType> CodecFromString(std::string_view name);
```

It is recommended to pass string_view by value.